### PR TITLE
Adds RBS signatures to help IDE to find usages and potential errors

### PR DESCRIPTION
**Pull request by pftg**
Adds RBS files signatures to help IDE to find the problems or find usage of the unstructured scalars. RubyMine do not need any tools to start use them.

After run, found several problems and added hot-fixes:
* There are unused code, which should generate runtime error if it will be executed
* Made explicit boolean returns
* Explicit driver options types to prepare to move into named params

### CODE REVIEW

**Comment by pftg** [comment]
@donv @UweKubosch please review. There will be more types declarations and cleanups.

File: lib/capybara/screenshot/diff/test_methods.rb [hunk 1]
```diff
@@ -94,13 +95,20 @@ def screenshot(name, skip_stack_frames: 0, **options)
         end
 
         def assert_image_not_changed(caller, name, comparison)
-          return unless comparison.different?
+          return nil unless comparison.different?
```
Comment (donv) [inline lib/capybara/screenshot/diff/test_methods.rb:98]: Why explicit `nil` return value?

File: lib/capybara/screenshot/diff/test_methods.rb [hunk 2]
```diff
@@ -56,6 +54,9 @@ def screenshot_group(name)
           FileUtils.rm_rf screenshot_dir
         end
 
+        # @param [(Symbol | String)] name
```
Comment (donv) [inline lib/capybara/screenshot/diff/test_methods.rb:57]: What effect do these kinds of comments have?

Comment (pftg) [reply lib/capybara/screenshot/diff/test_methods.rb:57]: yeah, this was the first attempt to document and help RubyMine to validate types from the YARD doc. But then moved to rbs files. Will cleanup this for now.

File: .gitignore [hunk 3]
```diff
@@ -12,3 +12,5 @@
 /tmp/
 /test/fixtures/**/*.committed.png
 /test/fixtures/**/*.latest.png
+/vendor/sigs/
+/*_*.png
```
[the remark this answers is not in the document]
Comment (pftg) [reply .gitignore:16]: yeah, we have tests that generated png files in the root dir. That was a fast workaround to make git clean. Will remove it. And later will add more cleaner fix.

**Comment by pftg** [comment]
@donv updated with fixes: removed yard docs and other notices. Please recheck

**Review by donv — APPROVED** [review]

**Pull request merged**